src/pages/MainPage/index.jsx

```diff
@@ -1,14 +1,77 @@
-import { QuoteCard } from '../../Components/QuoteCard'; 
-import { Button } from '../../Components/Button';
+import { QuoteCard } from "../../components/QuoteCard";
+import { Button } from "../../components/Button";
+import { useQuotesContext } from "../../QuotesContextProvider";
+import {
+  useQuoteIndexContext,
+  useQuoteIndexDispatchContext,
+} from "../../QuoteIndexContextProvider";
+import { useQuotesDispatchContext } from "../../QuotesContextProvider";
 
-export const MainPage = ({quote , author ,likeCount, handleNextQuoteClick , handleLikeQuoteClick}) =>{
-  return(
+export const MainPage = () => {
+  const dispatchQuotes = useQuotesDispatchContext();
+  const quotes = useQuotesContext();
+  const currentIndex = useQuoteIndexContext();
+  const dispatchQuoteIndex = useQuoteIndexDispatchContext();
+
+  function handleNextQuoteClick() {
+    const randomIndex = Math.floor(Math.random() * quotes.length);
+    dispatchQuoteIndex(randomIndex);
+  }
+  function handleLike() {
+    const newQuotes = quotes.map((quote, index) => {
+      if (currentIndex === index) {
+        return { ...quote, likeCount: quote.likeCount + 1 };
+      } else return quote;
+    });
+    dispatchQuotes(newQuotes);
+  }
+
+  function handleFavorite() {
+    const updatedQuotes = quotes.map((quote, index) => {
+      console.log(currentIndex);
+
+      if (currentIndex === index) {
+        return { ...quote, isFavorite: !quote.isFavorite };
+      } else return quote;
+    });
+    dispatchQuotes(updatedQuotes);
+  }
+
+  return (
     <main>
-      <h1>Random Quote Generator</h1>
-      <QuoteCard quote={quote} author={author} likeCount={likeCount} />
-      <Button label="Next Quote" handleOnClick={handleNextQuoteClick} className="btn"/>
-      <Button label="Like" handleOnClick={handleLikeQuoteClick} className="btn btn-like" />
-    </main>
+      <QuoteCard
+        quote={quotes[currentIndex].quote}
+        author={quotes[currentIndex].author}
+        likeCount={"Like : " + quotes[currentIndex].likeCount}
```

Comment:
You need to pass like count (number of likes) here, it does not make much sense to add "Like :" string as well - just move the text to the QuoteCard component.